### DISCUSSION TITLE
Error using nested Components with pagination

### DISCUSSION THREAD

**Discussion by SchuBu** [livewire/livewire#4434] [7. Bugs (archived)] [answered]
Hello there,

I'm trying to create a nested livewire component with pagination. I create a list of posts. Every post should contain another livewire component and receives the post-id as a parameter. But if I render my component in the browser there are issues: Elements don't show up correctly. The post-id I send to my nested component is not "refreshed" on paginating. I'm using keying the elements etc. I'll show you the code above. I think this is really a mess with livewire.

My solution - if this is really a bug - is to raise an event with the `TestTable` component and let the `TestItem` component listen to it. But this is just a workaround. Did I do anything wrong? I think it's a bug.

TestTable.class
```php
<?php
namespace App\Http\Livewire;
use App\Models\Post;
use Livewire\Component;
use Livewire\WithPagination;
class TestTable extends Component
{
    use WithPagination;
    public function render()
    {
        $post = Post::withTrashed()->paginate(15);
        return view('livewire.test-table', compact('post'));
    }
}

```

TestItem.php
```php
<?php
namespace App\Http\Livewire;
use Livewire\Component;
class TestItem extends Component
{
    public $pid;
    public function render()
    {
        return view('livewire.test-item');
    }
}
```

test-table.blade.php
```blade
<div>
    @foreach($post as $item)
        <div wire:key="my-element-{{ $loop->index }}">{{$item->id}}:
            <livewire:test-item :pid="$item->id" :wire:key="$loop->index"/>
        </div>
    @endforeach
    {{$post->links()}}
</div>
```

test-item.blade.php
```blade
<div>
    {{ $pid }}
</div>
```
This is the browser result on load:
![step1](https://user-images.githubusercontent.com/30638214/146926966-7926bfeb-4795-4125-9e7b-c30f8d06b242.png)

This is the result after I click on page "2"
![step2](https://user-images.githubusercontent.com/30638214/146927208-35e48f2d-07da-4fac-b0a3-74add2464fb5.png)

**Comment by jonneroelofs** [comment, marked as answer]
I think the problem is that you use the index as a key. But although the content of the row may change when you paginate the index will stay the same.

Try using {{$item->id}} as your key.

**Reply by SchuBu** [reply]
Thanks for the tip. I just realised, that using the "tag"-variant with ":wire:key" broke the component (wondering why ...). I replaced it by 

```blade
@livewire('test-item', ['pid' => $item->id], $loop->index)
```

and tested it also with:
```blade
@livewire('test-item', ['pid' => $item->id], $item->id)
```

same problem:
![step2](https://user-images.githubusercontent.com/30638214/146946195-ebb1dec1-562b-4a63-b4bf-22cae706a191.png)

**Reply by SchuBu** [reply]
Another thing: It seems that one click on the pagination link breaks the js ...
![2021-12-21_153553](https://user-images.githubusercontent.com/30638214/146947420-f36cafcf-49d9-4d1e-b8e5-c3a5ef720935.png)

**Reply by jonneroelofs** [reply]
Have a look here:
https://laravel-livewire.com/docs/2.x/nesting-components

You syntax is slightly different then in the docs. I never use the @livewire syntax, but I see that in the docs it has a key() function around the key.

Maybe also lose the extra div around test-item, just to be sure?

I would probably write it like:
```php
<div>
  <div>
      @foreach($post as $item)
            <livewire:test-item :pid="$item->id" :wire:key="$item->id"/>
      @endforeach
  
  </div>
  <div>
      {{$post->links()}}
  </div>
</div>
```
